Index: linux-2.6.29/drivers/dpm/proc.c
===================================================================
--- linux-2.6.29.orig/drivers/dpm/proc.c	2010-04-07 13:00:43.000000000 -0400
+++ linux-2.6.29/drivers/dpm/proc.c	2010-04-07 13:02:16.000000000 -0400
@@ -165,7 +165,7 @@
 	char *whitespace = " \t\r\n";
 	int ret = 0, ntoks;
 
-	if (current->uid != 0)
+	if (current->cred->uid != 0)
 		return -EACCES;
 	if (count == 0)
 		return 0;
Index: linux-2.6.29/drivers/net/wireless/marvell8686/wlan_fops.c
===================================================================
--- linux-2.6.29.orig/drivers/net/wireless/marvell8686/wlan_fops.c	2010-04-07 13:23:13.000000000 -0400
+++ linux-2.6.29/drivers/net/wireless/marvell8686/wlan_fops.c	2010-04-07 13:23:24.000000000 -0400
@@ -44,19 +44,19 @@
 
     /* Save uid and gid used for filesystem access.  */
 
-    orgfsuid = current->fsuid;
-    orgfsgid = current->fsgid;
+    orgfsuid = current->cred->fsuid;
+    orgfsgid = current->cred->fsgid;
 
     /* Set user and group to 0 (root) */
-    current->fsuid = 0;
-    current->fsgid = 0;
+    #warning current->cred->fsuid = 0;
+    #warning current->cred->fsgid = 0;
 
     /* Open the file in kernel mode */
     file_ret = filp_open(filename, flags, mode);
 
     /* Restore the uid and gid */
-    current->fsuid = orgfsuid;
-    current->fsgid = orgfsgid;
+    #warning current->cred->fsuid = orgfsuid;
+    #warning current->cred->fsgid = orgfsgid;
 
     /* Check if the file was opened successfully
        and return the file pointer of it was.  */
@@ -79,19 +79,19 @@
         return -ENOENT;
 
     /* Save uid and gid used for filesystem access.  */
-    orgfsuid = current->fsuid;
-    orgfsgid = current->fsgid;
+    orgfsuid = current->cred->fsuid;
+    orgfsgid = current->cred->fsgid;
 
     /* Set user and group to 0 (root) */
-    current->fsuid = 0;
-    current->fsgid = 0;
+    #warning current->cred->fsuid = 0;
+    #warning current->cred->fsgid = 0;
 
     /* Close the file in kernel mode (user_id = 0) */
     file_ret = filp_close(file_ptr, 0);
 
     /* Restore the uid and gid */
-    current->fsuid = orgfsuid;
-    current->fsgid = orgfsgid;
+    #warning current->cred->fsuid = orgfsuid;
+    #warning current->cred->fsgid = orgfsgid;
 
     return (file_ret);
 }
@@ -128,12 +128,12 @@
         return -EINVAL;
 
     /* Save uid and gid used for filesystem access.  */
-    orgfsuid = current->fsuid;
-    orgfsgid = current->fsgid;
+    orgfsuid = current->cred->fsuid;
+    orgfsgid = current->cred->fsgid;
 
     /* Set user and group to 0 (root) */
-    current->fsuid = 0;
-    current->fsgid = 0;
+    #warning current->cred->fsuid = 0;
+    #warning current->cred->fsgid = 0;
 
     /* Save FS register and set FS register to kernel
        space, needed for read and write to accept
@@ -150,8 +150,8 @@
     set_fs(orgfs);
 
     /* Restore the uid and gid */
-    current->fsuid = orgfsuid;
-    current->fsgid = orgfsgid;
+    #warning current->cred->fsuid = orgfsuid;
+    #warning current->cred->fsgid = orgfsgid;
 
     return (file_ret);
 }
