From 688a54ca500cc2bfa2ed301d95c969fc88b5f359 Mon Sep 17 00:00:00 2001
From: gqwang <gqwang@hhcn.com>
Date: Mon, 17 Aug 2009 14:46:57 +0800
Subject: [PATCH] Fix some bugs of framebuffer driver

---
 drivers/video/tccfb_a070vw04.c |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/drivers/video/tccfb_a070vw04.c b/drivers/video/tccfb_a070vw04.c
index dcab8fb..db2a6ba 100644
--- a/drivers/video/tccfb_a070vw04.c
+++ b/drivers/video/tccfb_a070vw04.c
@@ -3,7 +3,7 @@
 #include <linux/fb.h>
 #include <linux/delay.h>
 #include <linux/platform_device.h>
-#include <asm-arm/plat-tcc8900/gpio.h>
+#include <mach/gpio.h>
 
 #define LCD_SCEN    TCC_GPC29
 #define LCD_SCL	    TCC_GPC30
-- 
1.6.3.3

