From b403a9afc3d8ed6ecc6174e16885f7ed79a90c8b Mon Sep 17 00:00:00 2001
From: YouSheng <david.ysh@gmail.com>
Date: Fri, 6 Nov 2009 15:42:20 +0800
Subject: [PATCH] Fix after reboot vlc nosound problem

---
 sound/soc/soc-core.c |   10 ++++++----
 1 files changed, 6 insertions(+), 4 deletions(-)

diff --git a/sound/soc/soc-core.c b/sound/soc/soc-core.c
index 16c7453..b72561a 100644
--- a/sound/soc/soc-core.c
+++ b/sound/soc/soc-core.c
@@ -359,10 +359,12 @@ static int soc_codec_close(struct snd_pcm_substream *substream)
 	cpu_dai->runtime = NULL;
 
 	if (substream->stream == SNDRV_PCM_STREAM_PLAYBACK) {
-		/* start delayed pop wq here for playback streams */
-		codec_dai->pop_wait = 1;
-		schedule_delayed_work(&socdev->delayed_work,
-			msecs_to_jiffies(pmdown_time));
+		if (codec->bias_level == SND_SOC_BIAS_ON) { //XXX: FIX VLC
+			/* start delayed pop wq here for playback streams */
+			codec_dai->pop_wait = 1;
+			schedule_delayed_work(&socdev->delayed_work,
+					      msecs_to_jiffies(pmdown_time));
+		}
 	} else {
 		/* capture streams can be powered down now */
 		snd_soc_dapm_stream_event(codec,
-- 
1.6.3.3

