From 3f302191ee7fdad86f17542623b852ec318eb6af Mon Sep 17 00:00:00 2001
From: YouSheng <david.ysh@gmail.com>
Date: Fri, 13 Nov 2009 15:01:40 +0800
Subject: [PATCH] Reduce i2c freq 100->50

---
 arch/arm/mach-tcc8900/devices.c |    8 ++++----
 1 files changed, 4 insertions(+), 4 deletions(-)

diff --git a/arch/arm/mach-tcc8900/devices.c b/arch/arm/mach-tcc8900/devices.c
index d447274..da91853 100644
--- a/arch/arm/mach-tcc8900/devices.c
+++ b/arch/arm/mach-tcc8900/devices.c
@@ -191,8 +191,8 @@ static struct resource tcc8900_i2c_resources[] = {
 		.flags	= IORESOURCE_MEM,
     },
 	[1] = {
-		.start	= 100,					/* I2C ch0 100Kbps */
-		.end	= 100,					/* I2C ch1 100Kbps */
+		.start	= 50,					/* I2C ch0 100Kbps */
+		.end	= 50,					/* I2C ch1 100Kbps */
 		.flags	= IORESOURCE_MEM,
     },
 //	[1] = {
@@ -208,8 +208,8 @@ static struct resource tcc8900_i2c_resources[] = {
 		.flags	= IORESOURCE_MEM,
     },
 	[3] = {
-		.start	= 100,					/* SMU_I2C ch0 100Kbps */
-		.end	= 100,					/* SMU_I2C ch1 100Kbps */
+		.start	= 50,					/* SMU_I2C ch0 100Kbps */
+		.end	= 50,					/* SMU_I2C ch1 100Kbps */
 		.flags	= IORESOURCE_MEM,
     },
 };
-- 
1.6.3.3

