Index: linux-2.6.29/sound/soc/codecs/wm8987.c
===================================================================
--- linux-2.6.29.orig/sound/soc/codecs/wm8987.c	2010-04-07 15:02:04.000000000 -0400
+++ linux-2.6.29/sound/soc/codecs/wm8987.c	2010-04-07 15:46:02.000000000 -0400
@@ -613,8 +613,10 @@
 	return 0;
 }
 
-static int wm8987_pcm_hw_params(struct snd_pcm_substream *substream,
-	struct snd_pcm_hw_params *params)
+static int wm8987_pcm_hw_params(
+	struct snd_pcm_substream *substream,
+	struct snd_pcm_hw_params *params,
+	struct snd_soc_dai *dai)
 {
 	struct snd_soc_pcm_runtime *rtd = substream->private_data;
 	struct snd_soc_device *socdev = rtd->socdev;
@@ -693,6 +695,13 @@
 #define WM8987_FORMATS (SNDRV_PCM_FMTBIT_S16_LE | SNDRV_PCM_FMTBIT_S20_3LE |\
 	SNDRV_PCM_FMTBIT_S24_LE)
 
+static struct snd_soc_dai_ops wm8987_dai_ops = {
+	.hw_params     = wm8987_pcm_hw_params,
+	.digital_mute  = wm8987_mute,
+	.set_fmt       = wm8987_set_dai_fmt,
+	.set_sysclk    = wm8987_set_dai_sysclk,
+};
+
 struct snd_soc_dai wm8987_dai = {
 	.name = "WM8987",
 	.playback = {
@@ -707,14 +716,7 @@
 		.channels_max = 2,
 		.rates = WM8987_RATES,
 		.formats = WM8987_FORMATS,},
-	.ops = {
-		.hw_params = wm8987_pcm_hw_params,
-	},
-	.dai_ops = {
-		.digital_mute = wm8987_mute,
-		.set_fmt = wm8987_set_dai_fmt,
-		.set_sysclk = wm8987_set_dai_sysclk,
-	},
+	.ops = &wm8987_dai_ops,
 };
 EXPORT_SYMBOL_GPL(wm8987_dai);
 
@@ -829,7 +831,7 @@
 
 	wm8987_add_controls(codec);
 	wm8987_add_widgets(codec);
-	ret = snd_soc_register_card(socdev);
+	ret = snd_soc_init_card(socdev);
 	if (ret < 0) {
 		printk(KERN_ERR "wm8987: failed to register card\n");
 		goto card_err;
Index: linux-2.6.29/sound/soc/codecs/wm8510.c
===================================================================
--- linux-2.6.29.orig/sound/soc/codecs/wm8510.c	2010-04-07 16:42:38.000000000 -0400
+++ linux-2.6.29/sound/soc/codecs/wm8510.c	2010-04-07 16:43:19.000000000 -0400
@@ -321,7 +321,8 @@
 	Nmod = target % source;
 	Kpart = FIXED_PLL_SIZE * (long long)Nmod;
 
-	do_div(Kpart, source);
+	//do_div(Kpart, source);
+	div_s64(Kpart, source);
 
 	K = Kpart & 0xFFFFFFFF;
 
Index: linux-2.6.29/sound/soc/codecs/wm8753.c
===================================================================
--- linux-2.6.29.orig/sound/soc/codecs/wm8753.c	2010-04-07 16:43:43.000000000 -0400
+++ linux-2.6.29/sound/soc/codecs/wm8753.c	2010-04-07 16:43:48.000000000 -0400
@@ -727,7 +727,8 @@
 	Nmod = target % source;
 	Kpart = FIXED_PLL_SIZE * (long long)Nmod;
 
-	do_div(Kpart, source);
+	//do_div(Kpart, source);
+	div_s64(Kpart, source);
 
 	K = Kpart & 0xFFFFFFFF;
 
