From 4a7cd81fea97058c5feec34a357c525b8c89f7d9 Mon Sep 17 00:00:00 2001
From: gqwang2007 <gqwang@hhcn.com>
Date: Wed, 20 Jan 2010 17:26:02 +0800
Subject: [PATCH] Remove the click and pop noise when first open VLC

---
 sound/soc/codecs/wm8987.c |    9 ++++++++-
 1 files changed, 8 insertions(+), 1 deletions(-)

diff --git a/sound/soc/codecs/wm8987.c b/sound/soc/codecs/wm8987.c
index 1c04560..983950f 100755
--- a/sound/soc/codecs/wm8987.c
+++ b/sound/soc/codecs/wm8987.c
@@ -35,6 +35,7 @@
 
 static struct snd_soc_codec *p_codec = NULL;
 static struct delayed_work wm8987_unmute_work;
+static unsigned char wm8987_first_bias_on = 0;
 
 /* codec private data */
 struct wm8987_priv {
@@ -672,6 +673,8 @@ static void wm8987_delayed_unmute(struct work_struct* work)
 	else {
 		mute_reg = wm8987_read_reg_cache(p_codec, WM8987_ADCDAC) & 0xfff7;
 		wm8987_write(p_codec, WM8987_ADCDAC, mute_reg);
+		if(wm8987_first_bias_on)
+		    wm8987_first_bias_on = 0;
 	}
 }
 
@@ -684,7 +687,10 @@ static int wm8987_mute(struct snd_soc_dai *dai, int mute)
 		wm8987_write(codec, WM8987_ADCDAC, mute_reg | 0x8);
 	else if(mute_reg & 0x8) {
 		cancel_delayed_work(&wm8987_unmute_work);
-		schedule_delayed_work(&wm8987_unmute_work, msecs_to_jiffies(200));
+		if(wm8987_first_bias_on)
+		    schedule_delayed_work(&wm8987_unmute_work, msecs_to_jiffies(500));
+		else
+		    schedule_delayed_work(&wm8987_unmute_work, msecs_to_jiffies(200));
 	}
 
 	return 0;
@@ -697,6 +703,7 @@ static int wm8987_set_bias_level(struct snd_soc_codec *codec,
 
 	switch (level) {
 	case SND_SOC_BIAS_ON:
+		wm8987_first_bias_on = 1;
 		/* do nothing, because 'case SND_SOC_BIAS_PREPARE' has done everything */
 		break;
 	case SND_SOC_BIAS_PREPARE:
-- 
1.6.3.3

